### PR TITLE
fix: some turbo mode tests and some buffer issue

### PR DESCRIPTION
**Pull request by ajbt200128**

### CODE REVIEW

File: js/Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ test:
 	cd libpcre; $(MAKE) test
 	cd libyaml; $(MAKE) test
 	cd engine; $(MAKE) test
-	cd languages; $(MAKE) -j $(nproc) test # These are easily parallelizable
+	cd languages; $(MAKE) test # These are easily parallelizable
```
Comment (tpetr) [inline js/Makefile:16]: why remove the `-j`?

File: js/tests/Test_jsoo.ml [hunk 2]
```diff
@@ -23,10 +23,13 @@ open Semgrep_js_shared
 (* Code *)
 (*****************************************************************************)
 
+let filtered = [ "Cpp"; "C++"; "Julia"; "Ruby" ]
```
Comment (tpetr) [inline js/tests/Test_jsoo.ml:28]: can we add a comment re: why we're skipping these?

**Review by tpetr — APPROVED** [review]
2 nits but overall LGTM, thanks

File: js/Makefile (hunk 1 above)
Comment (ajbt200128) [reply js/Makefile:16]: Oops didn't mean to was trying something

File: js/tests/Test_jsoo.ml (hunk 2 above)
Comment (ajbt200128) [reply js/tests/Test_jsoo.ml:28]: Yup

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -133,43 +133,41 @@ jobs:
...
+    name: test semgrep js
+    needs: [build-semgrep-js-ocaml-test]
+    runs-on: ubuntu-22.04
+    container: emscripten/emsdk:3.1.46
+    env:
+      HOME: /root
+    steps:
+      - name: Restore git checkout cache
+        id: restore-git
+        uses: actions/cache/restore@v3
```
Comment (apt-itude) [inline .github/workflows/tests.yml:146]: I think this action only does the restore, but not the save. Is the cache being saved elsewhere? If not, I think the `actions/cache@v3` action will restore here and also trigger a save if the job completes successfully.

Comment (ajbt200128) [reply .github/workflows/tests.yml:146]: Cache is being saved in previous step

Comment (ajbt200128) [reply .github/workflows/tests.yml:146]: Wait no

Comment (ajbt200128) [reply .github/workflows/tests.yml:146]: Wait yes

Comment (apt-itude) [reply .github/workflows/tests.yml:146]: Ohh got it. I thought this was a cache just for this step, but I recall having a conversation about this last week.

**Review by apt-itude — APPROVED** [review]
I think this looks okay, although @spencerdrak has more experience with GHA and might have some wisdom to share about caching

**Pull request merged**